services/streammanager/kafka/kafkamanager.go

```diff
@@ -533,41 +544,27 @@ func (p *ProducerManager) Produce(jsonData json.RawMessage, destConfig interface
 		// return 400 if producer is invalid
 		return 400, "Could not create producer", "Could not create producer"
 	}
-
 	start := now()
 	defer func() { kafkaStats.produceTime.SendTiming(since(start)) }()
 
-	conf := configuration{}
-	jsonConfig, err := json.Marshal(destConfig)
-	if err != nil {
-		return makeErrorResponse(err) // returning 500 for retrying, in case of bad configuration
-	}
-	err = json.Unmarshal(jsonConfig, &conf)
-	if err != nil {
-		return makeErrorResponse(err) // returning 500 for retrying, in case of bad configuration
-	}
-
-	if conf.Topic == "" {
-		return makeErrorResponse(fmt.Errorf("invalid destination configuration: no topic"))
-	}
-
 	ctx, cancel := context.WithTimeout(context.TODO(), p.getTimeout())
 	defer cancel()
 	if kafkaBatchingEnabled {
-		return sendBatchedMessage(ctx, jsonData, p, conf.Topic)
+		return sendBatchedMessage(ctx, jsonData, p)
 	}
-	return sendMessage(ctx, jsonData, p, conf.Topic)
+
+	return sendMessage(ctx, jsonData, p)
```

Comment:
How do we guarantee that we are backwards compatible? We are totally ignoring the Topic in the destination's configuration...

I am personally afraid of relying exclusively to the transformer for setting the topic, without having a fallback in the server, since bugs in transformer are not uncommon. They way it seems now, a message will be aborted, aka lost, if transformer fails to populate the topic properly.

Reply:
I see your point we can fall back on the config topic here as well. As black box we can assume that transformer might not always send the topic as part of message

Reply:
I see your point @atzoum but I'm not sure that having some extra logic here just to safeguard us from a bug from another component (i.e. transformer) that we control is such a good idea. Can't we just have a test about this in the transformer codebase instead?

Reply:
I have done what @atzoum has asked but I haven't pushed it yet @koladilip @krishna2020 would like to hear your thoughts too

Reply:
@fracasula we need to be backwards compatible too with respect to the transformer versions we support

Reply:
If we have to be compatible with prior versions of the transformer then yeah it makes total sense :+1:

Reply:
This is addressed